core/conn/odbc/src/odbc/nsksrvrcore/srvrothers.cpp

```diff
@@ -4967,10 +4967,40 @@ odbc_SQLSvc_GetSQLCatalogs_sme_(
                     "and (ob.SCHEMA_NAME = '%s' or trim(ob.SCHEMA_NAME) LIKE '%s' ESCAPE '\\')  "
                     "and (ob.OBJECT_NAME = '%s' or trim(ob.OBJECT_NAME) LIKE '%s' ESCAPE '\\') "
                     "and (ob.OBJECT_TYPE in ('BT', 'VI')) "
-                    "and (trim(co.COLUMN_CLASS) not in('S', 'M'));",
+                    "and (trim(co.COLUMN_CLASS) not in('S', 'M')) "
+                    "union "
+                    "select "
+                    "cast('%s' as varchar(128)) TABLE_CAT, "
+                    "cast(trim(ob.SCHEMA_NAME) as varchar(128)) TABLE_SCHEM, "
+                    "cast(trim(ob.OBJECT_NAME) as varchar(128)) TABLE_NAME, "
+                    "cast(idx.is_unique as smallint) NON_UNIQUE, "
+                    "cast('' as varchar(128)) INDEX_QUALIFIER, "
+                    "cast(trim(ob_table.OBJECT_NAME) as varchar(128)) INDEX_NAME, "
+                    "cast(3 as smallint) TYPE, "
+                    "cast(0 as smallint) ORDINAL_POSITION, "
+                    "cast('' as varchar(128)) COLUMN_NAME, "
+                    "cast('' as char(1)) ASC_OR_DES, "
```

Comment:
I will change it to null. I will add the test to ODBC test. 
I think I still need add some new JDBC test. I will take some days.